File: Swift-C++-IR-Comparison/test/controlFlow/Swift/Guard.swift
@main
struct MyMain {

    static func main() {
        greet(person: ["name": "John"])
        // Prints "Hello John!"
        // Prints "I hope the weather is nice near you."
        greet(person: ["name": "Jane", "location": "Cupertino"])
    }
    static func greet(person: [String: String]) {
        guard let name = person["name"] else {
            return
        }

        print("Hello \(name)!")

        guard let location = person["location"] else {
            print("I hope the weather is nice near you.")
            return
        }

        print("I hope the weather is nice in \(location).")
    }
}

File: AutoTranspiledTests/try-0/Guard.cpp

struct MyMain {
  static void main(){
    greet(person={"name" : "John"});
	// Prints "Hello John!"
    // Prints "I hope the weather is nice near you."
    greet(person={"name" : "Jane", "location" : "Cupertino"}); 
  }
  static void greet(map<string,string> person) {
    if(!person.count("name")) {
      return;
    }
    cout << "Hello " << person["name"] << "!" << endl;
    if(!person.count("location")) {
      cout << "I hope the weather is nice near you." << endl;
      return;
    }
    cout << "I hope the weather is nice in " << person["location"] << "." << endl; 
  } 	
};